### ISSUE TITLE
[FEATURE] Support custom vector database

### ISSUE THREAD

**Issue by josx** [n4ze3m/page-assist#135]
In the place where i work [Cambá](www.camba.coop), we started to use page-assits to test our local  ai server with ollama.
Going deeper on the extension features we found KNOWLEDGE BASE.  
After that We realized that we also need to ask to multiple documents that are shared on the network.

So I think would be great if you can configure on KNOWLEDGE BASE another **provider** not just **chrome.storage** to get/save knowledge.  So I came up with the idea of using **client/server [Chroma](https://docs.trychroma.com/)** . 
With this new feature you can have your own knowledge server based on Chroma.

- Just config an URL on UI
- Make an alternative version of  [Vector.ts](https://github.com/n4ze3m/page-assist/blob/main/src/db/vector.ts) with chroma client API connecting to chroma Server.

What do you think about?

**Comment by n4ze3m** [comment]
Yes, I will add an option for a custom vector store in upcoming updates 😃

**Comment by abel87** [comment]
Is there anything update？I'm looking forward to this feature very much.
